lang/cs/Org.Apache.REEF.Network.Tests/NetworkService/WritableNetworkServiceTests.cs

```diff
@@ -0,0 +1,243 @@
+                    .BindNamedParameter<NetworkServiceOptions.NetworkServicePort, int>(
+                        GenericType<NetworkServiceOptions.NetworkServicePort>.Class,
+                        networkServicePort.ToString(CultureInfo.CurrentCulture))
+                    .BindNamedParameter<NamingConfigurationOptions.NameServerPort, int>(
+                        GenericType<NamingConfigurationOptions.NameServerPort>.Class,
+                        nameServicePort.ToString(CultureInfo.CurrentCulture))
+                    .BindNamedParameter<NamingConfigurationOptions.NameServerAddress, string>(
+                        GenericType<NamingConfigurationOptions.NameServerAddress>.Class,
+                        nameServiceAddr)
+                    .BindImplementation(GenericType<INameClient>.Class, GenericType<NameClient>.Class)
+                    .BindImplementation(GenericType<IObserver<WritableNsMessage<WritableString>>>.Class, GenericType<NetworkMessageHandler>.Class)
+                    .Build();
+
+                return TangFactory.GetTang().NewInjector(networkServiceConf).GetInstance<WritableNetworkService<WritableString>>();
+            }
+
+            var nameserverConf = TangFactory.GetTang().NewConfigurationBuilder()
+             .BindNamedParameter<NamingConfigurationOptions.NameServerPort, int>(
+                 GenericType<NamingConfigurationOptions.NameServerPort>.Class,
+                 nameServicePort.ToString(CultureInfo.CurrentCulture))
+             .BindNamedParameter<NamingConfigurationOptions.NameServerAddress, string>(
+                 GenericType<NamingConfigurationOptions.NameServerAddress>.Class,
+                 nameServiceAddr)
+             .BindImplementation(GenericType<INameClient>.Class, GenericType<NameClient>.Class)
+             .Build();
+            var injector = TangFactory.GetTang().NewInjector(nameserverConf);
+            var nameClient = injector.GetInstance<NameClient>();
+            var remoteManager = injector.GetInstance<WritableRemoteManagerFactory>();
+            return new WritableNetworkService<WritableString>(networkServicePort,
+                handler, factory, nameClient, remoteManager);
+        }
+
+        /// <summary>
+        /// The observer to handle incoming messages for WritableString
+        /// </summary>
+        private class MessageHandler : IObserver<WritableNsMessage<WritableString>>
+        {
+            private readonly BlockingCollection<WritableString> _queue;
+            private readonly IIdentifierFactory _factory;
+
+            public MessageHandler(BlockingCollection<WritableString> queue, IIdentifierFactory factory)
+            {
+                _queue = queue;
+                _factory = factory;
+            }
+
+            public void OnNext(WritableNsMessage<WritableString> value)
+            {
+                value.ConvertStringToIdentifier(_factory);
```

Comment:
Little bit wired to do it outside of the WritableNsMessage class. What id client doesn't know if he needs to do this conversion?